src/main/java/delete_post/use_case/DeletePostInteractor.java

```diff
@@ -1,37 +1,37 @@
 package delete_post.use_case;
 
 import delete_post.interface_adapters.DeletePostPresenter;
-import entities.User;
 
 public class DeletePostInteractor implements DeletePostInputBoundary{
 
-    private final DeletePostPresenter presenter;
+    private final DeletePostOutputBoundary outputBoundary;
     private final DeletePostDsGateway dataAccess;
 
-    public DeletePostInteractor(DeletePostPresenter presenter, DeletePostDsGateway dataAccess){
-        this.presenter = presenter;
+    public DeletePostInteractor(DeletePostOutputBoundary outputBoundary, DeletePostDsGateway dataAccess){
+        this.outputBoundary = outputBoundary;
         this.dataAccess = dataAccess;
     }
 
     @Override
     public void delete(DeletePostRequestModel requestModel){
-        if (requestModel.getIsTimer() || requestModel.getIsAdmin() ||
-                requestModel.getUser() == (requestModel.getPostUser())){
+        if (requestModel.getIsTimer()){
+            DeletePostResponseModel responseModel = new DeletePostResponseModel(requestModel.getPostId());
+
+            this.dataAccess.removeFavourites(requestModel.getPostId());
+            this.dataAccess.deletePost(requestModel.getPostId());
+            this.outputBoundary.prepareTimerView(responseModel);
```

Comment:
Minor detail: to reduce the amount of duplicate code, you could combine both if statements into one so that you have if(requestModel.getIsTimer() || requestModel.getIsAdmin() ||
                 requestModel.getUserId() == this.dataAccess.getPostUser(requestModel.getPostId())){
...
if(requestModel.getIsTimer()){
this.outputBoundary.prepareTimerView(responseModel);
}
else{
this.outputBoundary.prepareSuccessView(responseModel);
}}
Since the only difference between if it is the timer who calls delete post or if it is an admin/user is the view that is called in the end, right?